Refactor NetherlandsWithSchoolHolidays.get_christmas_holidays

jaraco opened the pull request:
The logic is more comprehensible using named functions and simple algebra instead of branchy/re-entrant logic and booleans.

jaraco commented:
I don't believe a changelog is needed for this internal refactor, but let me know if you'd like one.

brunobord commented:
yes, please add a changelog entry (and you'll need to rebase from master, but please wait for the 16.4.0 to be published)

jaraco commented:
Rebased and changelog added. Also, I removed the boolean parameters as they don't appear to be needed. Feel free to merge the parent commit instead if you think those parameters should be kept for compatibility.
